uberfire-services/uberfire-services-backend/src/test/java/org/guvnor/common/services/backend/metadata/MetadataServiceImplTest.java

```diff
@@ -75,6 +96,39 @@ public void testGetEmptyTagsNoOtherMetaView() {
                      tags.size());
     }
 
+    @Test
+    public void testSaveMetaData() throws IOException {
+        final InputStream is = spy(new ByteArrayInputStream("hello".getBytes(StandardCharsets.UTF_8)));
+        doReturn(is).when(ioService).newInputStream(any(),
+                                                    any());
+        final CommentedOption comment = new CommentedOption("comment");
+        doReturn(comment).when(commentedOptionFactory).makeCommentedOption("comment");
+        service.saveMetadata(path,
+                             new Metadata(),
+                             "comment");
+
+        verify(ioService).write(any(org.uberfire.java.nio.file.Path.class),
+                                any(byte[].class),
+                                anyMap(),
+                                eq(comment));
+        verify(is).close();
+    }
+
+    @Test(expected = GenericPortableException.class)
+    public void testSaveMetaDataException() {
+        final InputStream is = new ByteArrayInputStream("hello".getBytes(StandardCharsets.UTF_8));
```

Comment:
We need to close also this stream.

Reply:
Can't we really close the stream due to catching an exception, even if we use `org.assertj.core.api.Assertions.assertThatThrownBy()` ?

Reply:
But isn't the stream closed in `saveMetadata`?

Reply:
I added close, just incase something goes wrong before the close.